nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java

```diff
@@ -1152,9 +1153,18 @@ private boolean genericReturnIsNullable(MethodInvocationNode node) {
     if (node != null && config.isJSpecifyMode()) {
       MethodInvocationTree tree = node.getTree();
       if (tree != null) {
-        Nullness nullness =
-            genericsChecks.getGenericReturnNullnessAtInvocation(
-                ASTHelpers.getSymbol(tree), tree, node.getTreePath(), state);
+        // set flag that we are calling from dataflow to avoid infinite recursion, and verify that
+        // we are not already in a recursion.
+        verify(!genericsChecks.isCalledFromDataflow());
+        genericsChecks.setCalledFromDataflow(true);
```

Comment:
Ah, never mind my comment at https://github.com/uber/NullAway/pull/1309/files#diff-82d66c6ba15b87acc24509d7a3440bbe7606944197b929733d315fbe0a025435R113 below, I guess we are indeed using `GenericsChecks` as a re-usable singleton, not per invocation, so we can't make two instances with a `final calledFromDataflow`.

Threading the parameter might be a reasonable follow up, but I do see the description saying this was involved.